con-base/src/main/java/ba/codecentric/base/domain/Hotel.java

```diff
@@ -1,43 +1,65 @@
 package ba.codecentric.base.domain;
 
-import javax.persistence.*;
+import org.hibernate.validator.constraints.Email;
+import org.hibernate.validator.constraints.NotBlank;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
 
 @Entity
 public class Hotel {
 
     @Id
     @GeneratedValue
-    @Column(name = "id")
     private Integer id;
 
-    @Column(name = "name")
+    @NotBlank(message = "Please enter name of the hotel")
+    @Size(max = 128, message = "Name is too long, 128 characters allowed")
     private String name;
 
-    @Column(name = "rating")
+    @Max(value = 5, message = "Rating must be less than 5")
+    @Min(value = 1, message = "Rating must be greater than 0")
     private int rating;
 
-    @Column(name = "address")
+    @NotBlank(message = "Please enter street and number")
+    @Size(max = 128, message = "Address is too long, 128 characters allowed")
     private String address;
 
-    @Column(name = "zip")
+    @NotBlank(message = "Please enter zip code")
+    @Pattern(regexp = "^[0-9]{5}$", message = "Please enter a valid zip code e.g. 43452 (5 digit required)")
     private String zip;
```

Comment:
It could be done with proper formatting of components that display them. But I guess that leading zeroes are a good reason to leave as a varchar and save things from overengineering.